# Generated by Django 4.1.5 on 2023-01-20 17:18

from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('main', '0003_flight_price_flight_price_currency'),
    ]

    operations = [
        migrations.AddField(
            model_name='flight',
            name='max_reservations',
            field=models.IntegerField(default=120),
        ),
        migrations.AddField(
            model_name='flight',
            name='reservations',
            field=models.IntegerField(default=0),
        ),
        migrations.AddField(
            model_name='flight',
            name='reservators',
            field=models.ManyToManyField(blank=True, null=True, to=settings.AUTH_USER_MODEL),
        ),
        migrations.AddConstraint(
            model_name='flight',
            constraint=models.CheckConstraint(check=models.Q(('reservations__lte', models.F('max_reservations'))), name='can_reserve_check', violation_error_message='All seats have already been reserved.'),
        ),
    ]
